### DISCUSSION TITLE
Listen for any query changes and trigger component to rerender

### DISCUSSION THREAD

**Discussion by AsuraKev** [TanStack/query#7752] [Q&A]
Hi there,

I have a scenario where i am building breadcrumbs with react router, the information that a breadcrumb displays comes from the data fetched via react query. I am initialising a query inside react router loader like below:

![image](https://github.com/user-attachments/assets/4e50c92d-c321-42b2-99de-90696a1190c5)

As the nature of react router loader goes, it fires the loader as soon as the route mounts and in some cases I explicitly skip calling react query and the later on when the actual page component mounts a useQuery is fired. However this leaves the breadcrumb not aware about a useQuery being called.

In summary what i would like to do is that in the breadcrumb component, I would like to listen for all query state changes and rerender breadcrumb so that it can display the crumbs accordingly.

Any idea how to achieve this? I know in RTK Query I can just subscibe to query via useSelector, but not sure how to do it in react-query

thanks a lot

![image](https://github.com/user-attachments/assets/2f191e0b-2170-4d65-8169-535f480dccd5)

**Comment by TkDodo** [comment]
you would just subscribe with `useQuery`

**Reply by AsuraKev** [reply]
Thanks for the reply. Do you know if we can subscribe to all queries inside a component?

**Reply by TkDodo** [reply]
`queryClient.getQueryCache().subscribe`
